src/snowflake/cli/api/project/schemas/entities/application_entity_model.py

```diff
@@ -39,3 +41,21 @@ class ApplicationEntityModel(EntityModelBase):
         title="Whether to enable debug mode when using a named stage to create an application object",
         default=None,
     )
+
+    @field_validator("identifier")
+    @classmethod
+    def append_test_resource_suffix_to_identifier(
+        cls, input_value: Identifier | str
+    ) -> Identifier | str:
+        identifier = (
+            input_value.name if isinstance(input_value, Identifier) else input_value
+        )
+        with_suffix = append_test_resource_suffix(identifier)
+        if isinstance(input_value, Identifier):
+            data = dict(name=with_suffix)
+            if input_value.schema_:
+                data["schema"] = input_value.schema_
```

Comment:
any better way, not to have to remember all fields here? (scenario: someone adds a field in Identifier, but forgets to adjust here)?
- not sure if dumping the whole class dictionary or pydantic model and reconstructing it works (possibility 1)
- or we can introduce some kind of clone function within Identifier.

Reply:
Can you explain what you mean here? Identifier is [[db.]schema.].resource in SQL. What would be added in the future?

Reply:
I am not sure - but this defeats encapsulation as we are manually cloning an object with all its fields, and if in the future that object evolves, it's easy to miss. It seems a bit like an anti-pattern.

Reply:
Since Identifier is a Pydantic model, it's easy to do model_dump and then reload it.